cdap-docs/developers-manual/source/pipelines/developing-plugins/presentation-plugins.rst

```diff
@@ -0,0 +1,771 @@
+(``name``), *Dataset Base Path* (``basePath``), *Duration* (``duration``),
+and *Delay* (``delay``) could be grouped into a *Batch Source Configuration*.
+
+.. highlight:: json-ellipsis
+
+In the case of a *Batch Source* plugin, it could look like this::
+
+  {
+    "configuration-groups": [
+      {
+        "label": "Batch Source Configuration",
+        "properties": [
+          {
+            "name": "name",
+            "label": "Dataset Name",
+            ...
+          },
+          {
+            "name": "basePath",
+            "label": "Dataset Base Path",
+            ...
+          },
+          {
+            "name": "duration",
+            "label": "Duration",
+            ...
+          },
+          {
+            "name": "delay",
+            "label": "Delay",
+            ...
+          }
+        ]
+      }
+    ]
+    ...
+  }
+
+Once a group is established, you can configure how each of the individual properties
+inside the group is represented in the CDAP UI.
+
+.. _plugins-presentation-property-configuration:
+
+Property Configuration
+----------------------
+Each individual property of the plugin is represented by a configuration, composed of:
+
+- **name:** Name of the field (as supplied by the CDAP UI backend for the artifact).
+- **label:** Label to be displayed in the CDAP UI for the property.
+- :ref:`widget-type: <plugins-presentation-widgets>` The type of
+  widget to be used to represent this property.
+- **widget-attributes:** A map of attributes that the widget type requires to be defined in
+  order to render the property in the CDAP UI. The attributes required depend on the
+  widget type used.
+- :ref:`plugin-function: <plugins-presentation-plugin-function>`
+  An optional map of plugin method and its widget attributes that can be applied to a
+  particular plugin property.
+
+Note that with the exception of the value of the *label*, all properties and property
+values are case-sensitive.
+
+To find the available field names, you can use the :ref:`Artifact HTTP RESTful API 
+<http-restful-api-artifact>` to :ref:`retrieve plugin details 
+<http-restful-api-artifact-plugin-detail>` for an artifact, which will include all the
+available field names. (If the artifact is your own, you will already know the available
+field names from your source code.)
+
+In the case of a *Batch Source* plugin example, the ``configuration-groups`` could be
+represented by::
+
+  {
+    "configuration-groups": [
+      {
+        "label": "Batch Source",
+        "properties": [
+          {
+            "name": "name",
+            "label": "Dataset Name",
+            "widget-type": "dataset-selector"
+          },
+          {
+            "name": "basePath",
+            "label": "Dataset Base Path",
+            "widget-type": "textbox"
+          },
+          {
+            "name": "groupByFields",
+            "label": "Group By Fields",
+            "widget-type": "textbox",
+            "plugin-function": {
```

Comment:
I moved the explanation up from below the example to before it, and addressed why that `plugin-function` is there. It is defined in the list preceding the example as a possible option, which is why I wanted to include it.